src/data/rat/cast.lean

```diff
@@ -277,18 +277,15 @@ by simp [abs_eq_max_neg]
 
 end rat
 
-open rat ring_hom
+open rat
 
-lemma ring_hom.eq_rat_cast {k} [division_ring k] (f : ℚ →+* k) (r : ℚ) : f r = r :=
-calc f r = f (r.1 / r.2) : by rw [← int.cast_coe_nat, ← mk_eq_div, num_denom]
-     ... = f r.1 / f r.2 : f.map_div _ _
-     ... = r             : by rw [map_nat_cast, map_int_cast, cast_def]
-
--- This seems to be true for a `[char_p k]` too because `k'` must have the same characteristic
--- but the proof would be much longer
-@[simp] lemma map_rat_cast [division_ring α] [division_ring β] [char_zero α] [ring_hom_class F α β]
+@[simp] lemma map_rat_cast [division_ring α] [division_ring β] [ring_hom_class F α β]
   (f : F) (q : ℚ) : f q = q :=
-((f : α →+* β).comp $ cast_hom α).eq_rat_cast q
+calc f q = (f : α →+* β) q : rfl
+... = q : by rw [cast_def, ring_hom.map_div, (f : α →+* β).map_int_cast, map_nat_cast, cast_def]
```

Comment:
Why is this `: α →+* β` needed? Are we missing `ring_hom.map_div` for `ring_hom_class`?

Reply:
I guess `map_div` can't easily be generalised, right?

Reply:
We can add `map_inv_class` (or `inv_hom_class`) with two instances, one for `monoid_hom_class` from a `group`, and one for `monoid_with_zero_hom_class` from a `group_with_zero`.

Reply:
But IMHO, it is out of scope of this PR.

Reply:
What I mean for now is, can't we just add `ring_hom_class.map_div`? That way we don't need to insert the cast here.

Reply:
and yes, `data/int/cast` hasn't been fun-liked. I was going to do it after I did `nat/cast`, but I forgot. I may get around to it soon

Reply:
I'll do it as a part of this PR. I'm doing `map_inv` in a separate branch now.

Reply:
See #15985